test/spmd/checkpoint/test_dt_planner.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+# mypy: ignore-errors
+import os
+import shutil
+from typing import Dict, Union
+
+import torch
+import torch.distributed as dist
+import torch.distributed._shard.checkpoint as dist_cp
+from spmd import DeviceMesh, DTensor, Replicate
+from spmd import Shard as DShard
+from spmd import distribute_tensor
+from spmd.checkpoint.dt_planner import (
+    DistributedTensorLoadPlanner,
+    DistributedTensorSavePlanner,
+)
+from spmd.testing.common_utils import (
+    TEST_GPU_NUM,
+    DistTensorTestBase,
+    with_comms,
+)
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_utils import run_tests
+
+CHECKPOINT_DIR = "/scratch/checkpoint/test_dt_planner/"
```

Comment:
For a test suite we should use a temporary generated name. For example: https://github.com/pytorch/pytorch/pull/84952